### PR TITLE
export withOrchestration

### PR DESCRIPTION
**Pull request by turadg**
refs: #9757 
refs: https://github.com/Agoric/agoric-sdk/issues/9765

## Description

I noticed that `withOrchestration` isn't exported from the package. This exports it. It also exports the `OrchestrationTools` that the docs for that type require. I tried to not export `provideOrchestration` because it's more of an implementation detail, but JSDoc type expressions make it hard to export `OrchestrationTools` without also exporting `provideOrchestration`. I also considered getting rid of it but tests need to be able to set up OrchestrationTools without spinning up a contract. So I marked `provideOrchestration` as `@internal` so it doesn't show up in docs.

This also proposes a solution to,
- https://github.com/Agoric/agoric-sdk/issues/9765
using the [@interface](https://typedoc.org/tags/interface/#%40interface) tag to get the dynamic aspects to expand.

**Before**
```
CosmosInterchainService: ReturnType<[MakeCosmosInterchainService
```

**After**

```
interface CosmosInterchainService {
    [PASS_STYLE]: "remotable";
    [toStringTag]: string;
    __getInterfaceGuard__?: (() => undefined | InterfaceGuard<{
        makeAccount: any;
        provideICQConnection: any;
    }>);
    makeAccount(chainId, hostConnectionId, controllerConnectionId): Vow<IcaAccount>;
    provideICQConnection(controllerConnectionId): Guarded<{
        getLocalAddress(): `/ibc-port/${string}`;
        getRemoteAddress(): `/${string}ibc-port/${string}/ordered/${string}` | `/${string}ibc-port/${string}/unordered/${string}`;
        query(msgs): Promise<JsonSafe<ResponseQuery>[]>;
    }> | Vow<Guarded<{
        getLocalAddress(): `/ibc-port/${string}`;
        getRemoteAddress(): `/${string}ibc-port/${string}/ordered/${string}` | `/${string}ibc-port/${string}/unordered/${string}`;
        query(msgs): Promise<JsonSafe<ResponseQuery>[]>;
    }>>;
}
```

If that is agreeable I can convert the others in this PR or follow up.

### Testing Considerations
- `provideOrchestration` not in Markdown docs build for docs site
- `withOrchestration` signature is clear and concise

### CODE REVIEW

File: packages/orchestration/src/exos/cosmos-interchain-service.js [hunk 1]
```diff
@@ -43,13 +43,13 @@ const { Vow$ } = NetworkShape; // TODO #9611
...
 /**
- * @template {keyof OrchestrationPowers} K
+ * @template {keyof CosmosInterchainPowers} K
  * @param {PowerStore} powers
  * @param {K} name
  */
 const getPower = (powers, name) => {
   powers.has(name) || Fail`need powers.${b(name)} for this method`;
-  return /** @type {OrchestrationPowers[K]} */ (powers.get(name));
+  return /** @type {CosmosInterchainPowers[K]} */ (powers.get(name));
```
Comment (dckc) [inline packages/orchestration/src/exos/cosmos-interchain-service.js:52]: this `getPower` thing still isn't gone? darn.

**Comment by dckc** [comment]
I'd like to review this by looking at a cloudflare preview. Any idea why it's not here?

It was there on at least one recent PR: https://github.com/Agoric/agoric-sdk/pull/9798#issuecomment-2256752868

**Comment by 0xpatrickdev** [comment]
> I'd like to review this by looking at a cloudflare preview. Any idea why it's not here?
> 
> It was there on at least one recent PR: [#9798 (comment)](https://github.com/Agoric/agoric-sdk/pull/9798#issuecomment-2256752868)

Good idea. Clicking into Cloudflare Pages job I see:

Preview URL: | https://d7d9f03e.agoric-sdk.pages.dev
-- | --
Branch Preview URL: | https://ta-export-withorchestration.agoric-sdk.pages.dev

**Review by 0xpatrickdev — APPROVED** [review]
> /** @interface */

Nice finding!

Tangential to these changes, but [interfaces/_agoric_orchestration.Orchestrator](https://ta-export-withorchestration.agoric-sdk.pages.dev/interfaces/_agoric_orchestration.Orchestrator) is virtually unreadable with the KnownChains generic

**Comment by dckc** [comment]
> ... `Orchestrator` is virtually unreadable with the KnownChains generic

I noted that as a task in...

 - #9757

**Comment by dckc** [comment]
bleah. the preview suffers from...

 - #9729

File: packages/orchestration/src/utils/start-helper.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ import { makeZoeTools } from './zoe-tools.js';
  * @param {Baggage} baggage
  * @param {OrchestrationPowers} remotePowers
  * @param {Marshaller} marshaller
+ * @internal
```
Comment (dckc) [inline packages/orchestration/src/utils/start-helper.js:46]: doesn't seem to have the desired effect. I still see it:

image


https://ta-export-withorchestration.agoric-sdk.pages.dev/modules/_agoric_orchestration

**Review by dckc — APPROVED** [review]
this all looks like forward progress
in particular, https://ta-export-withorchestration.agoric-sdk.pages.dev/interfaces/_agoric_orchestration.CosmosInterchainService looks better.

some parts might not be working as well as expected, though

File: packages/orchestration/src/utils/start-helper.js (hunk 2 above)
Comment (turadg) [reply packages/orchestration/src/utils/start-helper.js:46]: Good catch. It needs [the excludeInternal option](https://typedoc.org/options/input/#excludeinternal), but even with that it's appearing. 
Screenshot 2024-07-31 at 8 33 18 AM

I'll wait for https://github.com/Agoric/agoric-sdk/pull/9727 and try again. I don't want to create merge conflicts for that one and maybe it fixes the bug.

Comment (turadg) [reply packages/orchestration/src/utils/start-helper.js:46]: I tried merging and that didn't fix it. Now I suspect a bug in typedoc

Comment (dckc) [reply packages/orchestration/src/utils/start-helper.js:46]: yeah, I've tried without luck to get `@internal` to do anything

**Comment by turadg** [comment]
Delivered better in https://github.com/Agoric/agoric-sdk/pull/10050

**Pull request closed without merging**